cmd/nodepool/core/create.go

```diff
@@ -184,3 +192,100 @@ func validateHostedClusterPayloadSupportsNodePoolCPUArch(ctx context.Context, cl
+func validMinorVersionCompatibility(ctx context.Context, client crclient.Client, name, namespace, nodePoolReleaseImage string, releaseProvider releaseinfo.Provider) error {
+	if nodePoolReleaseImage == "" {
+		return nil
+	}
+	logger := ctrl.LoggerFrom(ctx)
+
+	hcluster := &hyperv1.HostedCluster{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, hcluster); err != nil {
+		if !apierrors.IsNotFound(err) {
+			// For other errors (e.g. API server issues, RBAC problems), we should return the error
+			return fmt.Errorf("failed to get HostedCluster to check version compatibility: %w", err)
+		}
+
+		// This is expected to happen when we create a cluster since there is no created HostedCluster CR to check the
+		// payload from.
+		logger.Info("WARNING: failed to get HostedCluster to check version compatibility")
+		return nil
+	}
+
+	// Get the control plane version string
+	var controlPlaneVersionStr string
+	if len(hcluster.Status.Version.History) == 0 {
+		// If the cluster is in the process of installation, there is no history
+		// Use the desired version as the control plane version
+		controlPlaneVersionStr = hcluster.Status.Version.Desired.Version
+	} else {
+		// If the cluster is installed or upgrading
+		// Start with the most recent version from history as the default
+		controlPlaneVersionStr = hcluster.Status.Version.History[len(hcluster.Status.Version.History)-1].Version
+		// Update with any more recent Completed version if found
+		for _, history := range hcluster.Status.Version.History {
+			if history.State == "Completed" {
+				controlPlaneVersionStr = history.Version
+				break
+			}
+		}
+	}
+
+	// Parse control plane version
+	controlPlaneVersion, err := semver.Parse(controlPlaneVersionStr)
+	if err != nil {
+		return fmt.Errorf("parsing control plane version (%s): %w", controlPlaneVersionStr, err)
+	}
+
+	pullSecret := &corev1.Secret{}
+	if err = client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: hcluster.Spec.PullSecret.Name}, pullSecret); err != nil {
+		return fmt.Errorf("failed to get pull secret: %w", err)
+	}
+
+	releaseImage, err := releaseProvider.Lookup(ctx, nodePoolReleaseImage, pullSecret.Data[corev1.DockerConfigJsonKey])
+	if err != nil {
+		// Skip version check in disconnected environment where registry access is not available
+		logger.Info("WARNING: Unable to access the payload, skipping the Minor Version check.", "error", err.Error())
+		return nil
+	}
+
+	// Parse NodePool version
+	nodePoolVersion, err := semver.Parse(releaseImage.Version())
+	if err != nil {
+		return fmt.Errorf("parsing NodePool version (%s): %w", releaseImage.Version(), err)
+	}
+
+	// NodePool version cannot be higher than control plane version
+	if nodePoolVersion.Minor > controlPlaneVersion.Minor || nodePoolVersion.Major > controlPlaneVersion.Major {
+		return fmt.Errorf("NodePool version %d.%d cannot be higher than the HostedCluster version %d.%d",
```

Comment:
this still seems semantically incorrect, a hc with greater major but less minor will fail here with this message, and nested if/else become difficult to read.

Isn't this all we need?
```
// nodePool version must be the same or lower.
if nodePoolVersion.GT(controlPlaneVersion) {
   fail
}
```

Reply:
And when we want to allow greater z
```
	// NodePool version cannot be higher than control plane version
	// major version must be the same.
	if nodePoolVersion.Major != controlPlaneVersion.Major {
		fail
	}

	// minor version must be the same or lower.
	if nodePoolVersion.Minor > controlPlaneVersion.Minor {
		fail
	}
```